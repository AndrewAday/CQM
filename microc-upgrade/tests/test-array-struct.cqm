struct foo {
  float f;
  int i;
}

void bar(struct foo foo)
{
  foo.f = 4.5;
}

int main()
{
  struct foo[] arr;
  struct foo foo;
  arr = make(struct foo, 2);
  foo = arr[0];
  bar(foo);
  foo.i = 2;
  print_float(foo.f);
  print(foo.i);
  return 0;
}
