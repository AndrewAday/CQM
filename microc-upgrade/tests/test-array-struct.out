4.500000
2
